library/llm.py

```diff
@@ -6,28 +6,65 @@ def __init__(self, model_name, vdb: VDB):
         self.model = GPT4All(model_name)
         self.vdb = vdb
 
-    def query(self, question, max_tokens=50):
+    def query(self, question, key, context_limit = 5, max_tokens=500):
```

Comment:
Context limit is the number of emails retrieved. It's now settable through the ask API (with n=). Max tokens is defaulted to something that will allow it to fully answer the questions I've asked (it's been cut off with 50)